File: JCE/dsa/10a.c
#include <stdio.h>

int linearSearch(int arr[], int n, int x) {
    for (int i = 0; i < n; i++) {
        if (arr[i] == x) {
            return i;
        }
    }
    return -1;
}

int main() {
    int n, x;
    printf("Enter number of elements in array: ");
    scanf("%d", &n);

    int arr[n];
    for (int i = 0; i < n; i++) {
        printf("Enter number to be inserted in array: ");
        scanf("%d", &arr[i]);
    }

    printf("Enter the number to be searched: ");
    scanf("%d", &x);

    int found = linearSearch(arr, n, x);
    if (found == -1) {
        printf("Element not found\n");
    } else {
        printf("Element found at position: %d\n", found + 1);
    }

    return 0;
}

File: JCE/dsa/10a.py
def linear_search(arr, x):
    for i in range(len(arr)):
        if arr[i] == x:
            return i
    return -1

arr = []
n = int(input("Enter number of elements in array: "))
for i in range(n):
    num = int(input("Enter number to be inserted in array: "))
    arr.append(num)

x = int(input("Enter the number to be searched: "))
found = linear_search(arr, x)
if (found == -1):
    print("Element not found")
else:
    print("Element found at position : ", found + 1)
